### ISSUE TITLE
Create Dropout Strategies 

### ISSUE THREAD

**Issue by Gil32610** [Gil32610/cryptography-miniproject#13]
Oi @HeloisaJ !

Vamos mudar um pouco o direcionamento dessa atividade para não ser estritamente necessário utilizarmos decorators.

Agora será necessário apenas alterar o tipo de regularização no `DimensionalityReductionConv` como um parâmetro do construtor. e.g.:

```python
class DimensionalityReductionConv(nn.Module):
    def __init__(self, 
                 ... 
                 out_channels=60,
                 reg_type=None,       # None, 'spatial', or 'dropblock'
                 reg_prob=0.1,        # Shared probability parameter
                 block_size=7         # Specific to DropBlock
                 ):
        super().__init__()
        ...
        self.reg_layer = nn.Identity() # Default: Does nothing
        if reg_type == 'spatial':
            self.reg_layer = nn.Dropout2d(p=reg_prob)
        elif reg_type == 'dropblock':
            # Assuming you have a DropBlock2D class implemented
            self.reg_layer = DropBlock2D(drop_prob=reg_prob, block_size=block_size)
        ...
```

Posteriormente implementamos uma estratégia mais robusta, com padrões Decorator e Strategy, etc

**Comment by HeloisaJ** [comment]
Oi @Gil32610 !!

Fiz uma implementação dos Dropouts utilizando Strategy e adicionei ao gbrasnet de forma semelhante ao código dessa issue. Eu queria conseguir testar de uma forma fácil o código, mas eu não sei como. Enfim, mas acho que está tudo certo, depois poderia dar uma conferida no código ?

Obrigada !!
